toolchain/lex/lex.cpp

```diff
@@ -116,8 +116,12 @@ class [[clang::internal_linkage]] Lexer {
   // Add a lexed token to the tokenized buffer, and reset any token-specific
   // state tracked in the lexer for the next token.
   auto AddLexedToken(TokenInfo info) -> TokenIndex {
+    TokenIndex token(buffer_.token_infos_.size());
+    buffer_.token_infos_.push_back(info);
+    buffer_.expected_max_parse_tree_size_ +=
+        info.kind().expected_max_parse_tree_size();
     has_leading_space_ = false;
-    return buffer_.AddToken(info);
```

Comment:
Why wasn't it sufficient to move the definition of this function to tokenized_buffer.h?

Reply:
It would have been, but it didn't seem worth having the function broken out given that all the code around the call to this was already directly manipulating the private members of the tokenized buffer. This is purely about what is nicer code wise, not a performance thing. The only performance thing was getting it out of the source file.

Reply:
I'd rather have a method, if just to reduce the duplicated logic.

Reply:
Ok, switched back. I had found it more confusing to have the mixture of methods manipulating state and direct code manipulations, particularly below in the error recovery code, but can keep it if preferred.